Assets/Scripts/World/Units/Enemy/Main/EnemyController.cs

```diff
@@ -125,9 +125,7 @@ public class EnemyController : UnitController<IEnemy, Enemy, EnemyList>, IEnemyC
 
     public void KillAllEnemies() {
         for (int i = 0; i < activeEnemies.Count; i++) {
-            int EnemyMaxHealth = activeEnemies.ElementAt(i).IMaxHealth;
-
-            activeEnemies.ElementAt(i).Heal(-EnemyMaxHealth);
+            activeEnemies.ElementAt(i).Destroy();
             RemoveActiveEnemy(activeEnemies.ElementAt(i));
```

Comment:
Looking good on the destroy method. The last thing I'm seeing is that we probably don't want to be modifying the HashSet as we're looping through it.

Instead, can we called activeEnemies.Clear(); after the for loop is complete.